lightning/src/ln/chan_utils.rs

```diff
@@ -551,11 +557,445 @@ pub fn build_htlc_transaction(prev_hash: &Txid, feerate_per_kw: u32, contest_del
 	}
 }
 
+/// Per-channel data used to build transactions in conjunction with the per-commitment data (CommitmentTransaction).
+///
+/// Normally, this is converted to the owner-independent DirectedChannelTransactionParameters
+/// before use, via the to_directed function.
+#[derive(Clone)]
+pub struct ChannelTransactionParameters {
+	/// Holder public keys
+	pub holder_pubkeys: ChannelPublicKeys,
+	/// Counter-party public keys
+	pub counterparty_pubkeys: ChannelPublicKeys,
+	/// The contest delay selected by the holder, which applies to counterparty-broadcast transactions
+	pub holder_selected_contest_delay: u16,
+	/// The contest delay selected by the counterparty, which applies to holder-broadcast transactions
+	pub counterparty_selected_contest_delay: u16,
+	/// The funding outpoint
+	pub funding_outpoint: OutPoint,
+	/// Whether the holder is the initiator of this channel.
+	/// This is an input to the commitment number obscure factor computation.
+	pub is_outbound_from_holder: bool,
+}
+
+impl ChannelTransactionParameters {
+	/// Convert the owner specific parameters to owner independent parameters
+	pub fn to_directed(&self, broadcaster_is_holder: bool) -> DirectedChannelTransactionParameters {
+		DirectedChannelTransactionParameters {
+			inner: self,
+			broadcaster_is_holder
+		}
+	}
+}
+
+impl_writeable!(ChannelTransactionParameters, 0, {
+	holder_pubkeys,
+	counterparty_pubkeys,
+	holder_selected_contest_delay,
+	counterparty_selected_contest_delay,
+	funding_outpoint,
+	is_outbound_from_holder
+});
+
+/// Static channel fields used to build transactions given per-commitment fields (CommitmentTransaction).
+///
+/// If holder is the broadcaster, we map structure elements
+/// (holder_pubkeys, counterparty_pubkeys, counterparty_selected_contest_delay) to
+/// (broadcaster_pubkeys, countersignatory_pubkeys, contest_delay).  If the counterparty is the
+/// broadcaster, the parties are swapped in the above mapping.
+///
+/// This structure makes transaction fields (as consumed by CommitmentTransaction) independent of
+/// transaction ownership.
```

Comment:
This is a bit too much implementation detail for public documentation. Maybe just succinctly write this as:

> Static channel fields used to build transactions given per-commitment fields, independent of transaction ownership.

Perhaps clarifying that this is from the perspective of the broadcaster.

Reply:
@ariard - I think you had an opinion here...

Reply:
I'm fine with Jeff suggestion if he does find it's informative enough. I did call for a verbose comment in case of contributors not familiar with the (holder/counterparty) and (broadcaster/countersignatory) nomenclature.

Sometimes, it's hard to agree on a level of comment, we've all different depths of knowledge about this code part...

Reply:
I took Jeff's suggestion, and added a short comment about how to obtain this from the owner-specific struct, so the API user doesn't have to search for it.